cmd/juju/status/output_tabular.go

```diff
@@ -162,8 +162,29 @@ func printApplications(tw *ansiterm.TabWriter, fs formattedStatus) {
 				}
 			}
 			parts := strings.Split(version, "/")
+			// Charms deployed from the rocks or jujucharm repos have the
+			// image path set to <repo-url>/<user>/<charm_name>/<resource_name>. Charm name
+			// is somewhat redundant and takes up value space. So we'll replace
+			// with "res:" to indicate the named resource comes from the store repo.
+			fromJujuRepo := strings.Contains(app.Version, "jujucharms.") || strings.Contains(app.Version, "rocks.")
+			if fromJujuRepo && len(parts) > 2 && (len(version) > maxVersionWidth || parts[1] == app.Charm) {
```

Comment:
I really think we should attempt wherever we can, to pare down complexity, which would tend away from adding this more logic at this level of indentation to an already large method.